### ISSUE TITLE
global: jinja2hacks are causing troubles on Python 2.7.9

### ISSUE THREAD

**Issue by jirikuncar** [inveniosoftware/invenio#2862]
see for example https://travis-ci.org/inveniosoftware/invenio/builds/53365855

#### Workaround

Disable `invenio.ext.jinja2hacks` in `EXTENSIONS`.

--
cc @hachreak

**Comment by lnielsen** [comment]
I have the same problem but doesn't seem to be solely related to ``jinja2hacks``:
https://travis-ci.org/zenodo/zenodo#L6092

Could be the encoding of one specific file that causes the problem as described on https://github.com/mitsuhiko/markupsafe/issues/19

**Comment by hachreak** [comment]
```bash
file -bi invenio/base/templates/page_base.html
text/html; charset=us-ascii
```

**Comment by tiborsimko** [comment]
`page_base.html` loads stuff from `_macros.html` that contains UTF-8 character "×" in the data dismiss alert.  

You can perhaps try to replace it with regular ASCII "x" too see whether it helps?

P.S. Note also that elsewhere we do stuff with this character, such as replacing it in widgets...

```console
$ git grep -Hni × invenio
invenio/base/templates/_macros.html:25:        <a class="close" data-dismiss="alert" href="#">×</a>
invenio/modules/accounts/static/js/accounts/widgets.js:108:                        title = widget.find("h4").text().replace("×", " ").trim()
invenio/modules/comments/templates/comments/add_base.html:60:      <a class="close" data-dismiss="modal">×</a>
invenio/modules/comments/templates/comments/add_review_base.html:53:      <a class="close" data-dismiss="modal">×</a>
invenio/modules/communities/templates/communities/new_base.html:32:        <button type="button" class="close" data-dismiss="modal" aria-hidden="true">×</button>
invenio/modules/previewer/templates/previewer/bottom_base.html:27:    <button type="button" class="close" data-dismiss="modal" aria-hidden="true">×</button>
invenio/modules/search/static/js/search/facet.js:357:    close: $('<a/>', {'class': 'close', html:'&nbsp;×'}),
invenio/modules/workflows/templates/workflows/details_base.html:282:<a class="close" data-dismiss="alert">×</a>
[...]
```

**Comment by jirikuncar** [comment]
```console
$ find invenio -name \*.html | xargs pcregrep "[\x80-\xFF]"
invenio/modules/annotations/templates/annotations/notes_fragment_base.html:                <a class="pull-right" title="{{ _('Permalink to this note') }}" href="#{{ n.id }}">¶</a>
invenio/modules/comments/templates/comments/comments_base.html:              ¶
invenio/modules/comments/templates/comments/reviews_base.html:      <a class="pull-right" title="{{ _('Permalink to this review') }}" href="#{{ c.id }}">¶</a>
invenio/modules/communities/templates/communities/index_base.html:              <a title="first" href="{{ url_for('communities.index', **new_args) }}">«</a>
invenio/modules/communities/templates/communities/index_base.html:              <a title="last" href="{{ url_for('communities.index', **new_args) }}">»</a>
invenio/modules/previewer/templates/previewer/pdf_base.html:                <button class="btn btn-default" id="frst-page">«</button>
invenio/modules/previewer/templates/previewer/pdf_base.html:                <button class="btn btn-default" id="prev-page">‹</button>
invenio/modules/previewer/templates/previewer/pdf_base.html:                <button class="btn btn-default" id="next-page">›</button>
invenio/modules/previewer/templates/previewer/pdf_base.html:                <button class="btn btn-default" id="last-page">»</button>
invenio/modules/previewer/templates/previewer/pdfjs_base.html:            <button id="documentProperties" class="secondaryToolbarButton documentProperties" title="Document Properties…" tabindex="28" data-l10n-id="document_properties">
invenio/modules/previewer/templates/previewer/pdfjs_base.html:              <span data-l10n-id="document_properties_label">Document Properties…</span>
invenio/modules/search/templates/search/results_base.html:      <a title="first" href="{{ url_for('search.search', **args)+hash_tag }}">«</a>
invenio/modules/search/templates/search/results_base.html:      <a title="last" href="{{ url_for('search.search', **args)+hash_tag }}">»</a>
invenio/modules/search/templates/search/search_base.html:              <strong>{{ _('Filtering') }} …</strong>
```

**Comment by jalavik** [comment]
Some of these characters should be changed to the htmlentity, like `&times;` for the x and `&laquo;` and `&raquo;` for the « and » respectively.

Easy lookup: http://entity-lookup.leftlogic.com/

**Comment by hachreak** [comment]
I opened the row where the [error is generated](https://github.com/mitsuhiko/markupsafe/blob/master/markupsafe/__init__.py#L74)

```python
def __new__(cls, base=u'', encoding=None, errors='strict'):
  if hasattr(base, '__html__'):
    base = base.__html__()
  if encoding is None:
    return text_type.__new__(cls, base)
  return text_type.__new__(cls, base, encoding, errors)
```

If you open the definition of [text_type](https://github.com/mitsuhiko/markupsafe/blob/master/markupsafe/_compat.py#L15): it makes an assumption depending of python version (2 or 3).
Maybe now this assumption is not more true...? :P 

```python
if not PY2:
  text_type = str
else:
  text_type = unicode
```

Error: 
```python
TypeError: object.__new__(Markup) is not safe, use unicode.__new__()
```

**Comment by hachreak** [comment]
I discovered how reproduce the error:
 1. install a new fresh environment with invenio devscripts or remove `/home/vagrant/.virtualenvs/invenio2/src/invenio/.eggs/` directory.
 2. run `python setup.py test`

Note: the second time you run tests, the error magically disappear!

I was inspired by this issue: https://bugs.launchpad.net/pbr/+bug/1384919
And I found that if you previously install `unittest2`, the error disappear.
